ocaml/typing/includemod_errorprinter.ml

```diff
@@ -576,7 +570,7 @@ let dwith_context ?loc ctx printer =
   Location.msg ?loc "%a%t" Context.pp (List.rev ctx) printer
 
 let dwith_context_and_elision ?loc ctx printer diff =
-  if is_big (diff.got,diff.expected) then
+  if is_big (printer diff) then
     Location.msg ?loc "..."
   else
     dwith_context ?loc ctx (printer diff)
```

Comment:
It looks like we're calling `printer diff` twice now. I don't think it's actually expensive -- it's the partial application of a function -- but one could be misled by looking at the code. I suggest something like:

```ocaml
let print_diff ppf = printer diff ppf in
if is_big print_diff then
  ...
else
  ...
```

This better suggests that it's the use sites of `print_diff` that might be expensive, not the creation site.

Reply:
Good point - I lift the binding higher so it's clearer.